gmail/modules/listener/http_service.bal

```diff
@@ -22,29 +22,35 @@ service class HttpService {
 
     private gmail:Client gmailClient;
     public  string startHistoryId = "";
+    private string subscriptionResource;
     private Dispatcher dispatcher;
 
     public isolated function init(SimpleHttpService|HttpService httpService, gmail:Client gmailClient, 
-        string historyId) {
+        string historyId, string subscriptionResource) {
         self.gmailClient = gmailClient;
         self.startHistoryId = historyId;
+        self.subscriptionResource = subscriptionResource;
         self.dispatcher = new (httpService, self.gmailClient);
     }
 
     isolated resource function post mailboxChanges(http:Caller caller, http:Request request) returns @tainted error? {
+        json ReqPayload = check request.getJsonPayload();
+        string incomingSubscription = check ReqPayload.subscription;
         check caller->respond(http:STATUS_OK);
-        var  mailboxHistoryPage =  self.gmailClient->listHistory(self.startHistoryId);
-        
-        if (mailboxHistoryPage is gmail:MailboxHistoryPage) {
-            self.startHistoryId = mailboxHistoryPage.historyId;
-            log:printDebug("Next History ID = "+self.startHistoryId);
-            check self.dispatcher.dispatch(mailboxHistoryPage);                          
-        } else {
-            log:printError("Error occured while getting history.", 'error= mailboxHistoryPage);
-        }
-    }
 
-    public isolated function setStartHistoryId(string startHistoryId) {
-        self.startHistoryId = startHistoryId;
+        if (self.subscriptionResource === incomingSubscription) {
```

Comment:
What are we checking here? Should there be an `else` block to this? If it goes to elase block user will not know. If going to `else` is an error scenario, we need to log it.

Reply:
We create a subscription for pubsub topic internally in the listener and set the listener endpoint as push endpoint URL . When a change happens we will receive a push notification to this endpoint. If some one else configure the same url for a different subscription, we will get those push notifications also to this listener endpoint. To avoid that we check  the subscription resource which is in the push notification request is same to the subscription which we created. This is what checked here. If it true only Listener needs to proceeds further. So we can ignore the else part. But If we want to notify the Listener owner that someone else configured listener's endpoint to their subscription , we can log some message in the else part. WDYT ? Do we need to add that part ? 

https://developers.google.com/gmail/api/guides/push#receiving_notifications
https://cloud.google.com/pubsub/docs/push

Reply:
As we discussed in offline, added warning log in else part .